redfish-core/lib/pcie_slots.hpp

```diff
@@ -29,6 +32,88 @@
 namespace redfish
 {
 
+inline void doLinkAssociatedDiskBackplaneToChassis(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisId, const std::string& drivePath, size_t index,
+    const std::optional<std::string>& validChassisPath,
+    const std::vector<std::string>& assemblyList)
+{
+    if (!validChassisPath || assemblyList.empty())
+    {
+        BMCWEB_LOG_ERROR("Chassis not found");
+        messages::resourceNotFound(asyncResp->res, "Chassis", chassisId);
+        return;
+    }
+
+    auto it = std::find(assemblyList.begin(), assemblyList.end(), drivePath);
+    if (it == assemblyList.end())
+    {
+        BMCWEB_LOG_ERROR("Drive path {} not found in the assembly list",
+                         drivePath);
+        messages::internalError(asyncResp->res);
+    }
+
+    nlohmann::json::object_t item;
+    item["@odata.id"] = boost::urls::format(
+        "/redfish/v1/Chassis/{}/Assembly#/Assemblies/{}", chassisId,
+        std::to_string(it - assemblyList.begin()));
+
+    asyncResp->res.jsonValue["Slots"][index]["Links"]["Oem"]["IBM"]
```

Comment:
Should this also be ["AssociatedAssembly"] as for the actual item?

Reply:
It is building the content for ` ["AssociatedAssembly"] ` and it is set to it.

```
    nlohmann::json::object_t item;
    item["@odata.id"] = boost::urls::format(
        "/redfish/v1/Chassis/{}/Assembly#/Assemblies/{}", chassisId,
        std::to_string(it - assemblyList.begin()));

    asyncResp->res.jsonValue["Slots"][index]["Links"]["Oem"]["IBM"]
                            ["@odata.type"] = "#OemPCIeSlots.v1_0_0.PCIeLinks";
    asyncResp->res.jsonValue["Slots"][index]["Links"]["Oem"]["IBM"]
                            ["AssociatedAssembly"] = std::move(item);   <----
```

Json output will look like
```
                "AssociatedAssembly": {
                  "@odata.id": "/redfish/v1/Chassis/chassis/Assembly#/Assemblies/24"
                }
```